ydb/docs/en/core/yql/reference/types/serial.md

```diff
@@ -0,0 +1,95 @@
+
+## Description
+
+When a column of a serial type is defined, a separate schema object called a `Sequence` is created and bound to this column. This object is a private sequence generator and it is hidden from the user. The `Sequence` will be destroyed together with the table.
+
+The `Sequence` object supports several parameters that determine its behavior. These parameters can be altered after creation using the [ALTER SEQUENCE](../syntax/alter-sequence.md) command.
+
+By default, values generated by the `Sequence` start from one, are incremented by one with each new value, and are limited according to the chosen type.
+
+{% note info %}
+
+Serial columns are supported both for columns included in the primary key and for non-key columns.
+
+However, such columns cannot be [altered](../syntax/alter_table/family#mod-column-groups) or [dropped](../syntax/alter_table/columns.md) from the table — attempting to perform these operations will result in an error.
+
+{% endnote %}
+
+| Type        | Maximum Value         | YDB Type |
+|-------------|----------------------|----------|
+| SmallSerial | 2^15–1                | Int16    |
+| Serial2     | 2^15–1                | Int16    |
+| Serial      | 2^31–1                | Int32    |
+| Serial4     | 2^31–1                | Int32    |
+| Serial8     | 2^63–1                | Int64    |
+| BigSerial   | 2^63–1                | Int64    |
+
+If the sequence reaches its maximum value, insertion will result in an error:
+
+```text
+Error: Failed to get next val for sequence: /dev/test/users/_serial_column_user_id, status: SCHEME_ERROR
+    <main>: Error: sequence [OwnerId: <some>, LocalPathId: <some>] doesn't have any more values available, code: 200503
+```
+
+{% note info %}
+
+The next value is allocated by the generator before the actual insertion into the table and is considered used even if the row is not successfully inserted (for example, in case of transaction rollback).
+As a result, the values in such a column may have gaps and may not form a continuous sequence.
+
+{% endnote %}
+
+Tables with `Serial` columns support [copy](../../../reference/ydb-cli/tools-copy.md), [rename](../../../reference/ydb-cli/commands/tools/rename.md), [dump](../../../reference/ydb-cli/export-import/tools-dump.md), [restore](../../../reference/ydb-cli/export-import/import-file.md), and [import](../../../reference/ydb-cli/export-import/import-s3.md)/[export](../../../reference/ydb-cli/export-import/export-s3.md) operations.
+
+## Usage Example
+
+You should carefully choose the columns for your [PRIMARY KEY](../../../dev/primary-key/row-oriented.md). For scalability and high performance, you should avoid writing rows with monotonically increasing primary keys. In this case, all records will go to the last partition, and all the load will target a single server.
```

Comment:
Let's move this explanation on correctly choosing a primary key for the optimal performance to the article in `dev/` that's being linked as a separate section, then link to that specific section from here.

Hello-world style example and a link to read up on choosing a proper primary key should suffice for this article.